src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TornadoService.java

```diff
@@ -256,11 +258,22 @@ public DocumentInfo generateEt1VettingDocument(CaseData caseData, String userTok
         }
     }
 
-    private void buildEt1VettingInstruction(HttpURLConnection connection, CaseData caseData)
+    private void buildDocumentInstruction(HttpURLConnection connection, CaseData caseData, String documentName)
             throws IOException {
         try (OutputStreamWriter outputStreamWriter = new OutputStreamWriter(connection.getOutputStream(),
                 StandardCharsets.UTF_8)) {
-            outputStreamWriter.write(Et1VettingHelper.getDocumentRequest(caseData, tornadoConnection.getAccessKey()));
+            switch (documentName) {
+                case "ET1 Vetting.pdf" :
+                    outputStreamWriter.write(Et1VettingHelper.getDocumentRequest(caseData,
+                            tornadoConnection.getAccessKey()));
+                    break;
+                case "ET3 Processing.pdf" :
+                    outputStreamWriter.write(Et3VettingHelper.getDocumentRequest(caseData,
+                            tornadoConnection.getAccessKey()));
+                    break;
+                default:
+                    throw new NotFoundException("Document not found");
```

Comment:
This is the wrong exception to throw. It's a webjars exception and nothing to do with what the code is doing. How about
`throw new IllegalArgumentException("Unexpected document name " + documentName);`

Reply:
Also you should check the documentName argument at the start of the method

Reply:
Corrected exception and added one at the start of the method - clarification on the duplication if possible?

Reply:
I think I know what you mean. Both methods are creating the full request string but you think there's duplication over some properties? Out of the 4 top level fields, only 1 is the same - the access key, the rest change as it's specific to each document. I think that's what you mean, if not you'll need to clarify

Reply:
Duplication removed